examples/recording_service/pluggable_storage/cpp/FileStorageReader.hpp

```diff
@@ -0,0 +1,178 @@
+ * called by RTI Recording Service to create your class.
+ */
+RTI_RECORDING_STORAGE_READER_CREATE_DECL(FileStorageReader);
+
+/**
+ * This class acts as a factory for objects of classes FileStorageStreamReader
+ * and FileStorageStreamInfoReader. These objects are used by Replay and/or
+ * Converter to retrieve data from the storage.
+ */
+class FileStorageReader :
+        public rti::recording::storage::StorageReader {
+public:
+    FileStorageReader(const rti::routing::PropertySet& properties);
+    virtual ~FileStorageReader();
+
+    virtual rti::recording::storage::StorageStreamInfoReader *
+            create_stream_info_reader(
+                    const rti::routing::PropertySet& properties);
+
+    virtual void delete_stream_info_reader(
+            rti::recording::storage::StorageStreamInfoReader *stream_info_reader);
+
+    virtual rti::recording::storage::StorageStreamReader * create_stream_reader(
+            const rti::routing::StreamInfo& stream_info,
+            const rti::routing::PropertySet& properties);
+
+    virtual void delete_stream_reader(
+            rti::recording::storage::StorageStreamReader *stream_reader);
+private:
+    std::ifstream info_file_;
+    std::ifstream data_file_;
+    std::string file_name_;
+};
+
+/*
+ * This class uses the provided DynamicData specialization of a
+ * StorageStreamReader to read data stored by the classes in file
+ * FileStorageWriter.hpp, for a given data stream (DDS topic).
+ * All data is stored in a text file. Data is serialized to a text format.
+ * Note: this example is only fit to work with the provided type definition
+ * (type HelloMsg - see HelloMsg.idl for more information).
+ */
+class FileStorageStreamReader :
+        public rti::recording::storage::DynamicDataStorageStreamReader {
+public:
+
+    FileStorageStreamReader(std::ifstream *data_file);
+
+    virtual ~FileStorageStreamReader();
+
+    /*
+     * Implementation of the read operation. It should interpret the selector
+     * state object that expresses the specific needs of Replay/Converter about
+     * the data to be provided (data not read before vs data of any kind, lower
+     * and upper time limits, etc).
+     */
+    virtual void read(
+            std::vector<dds::core::xtypes::DynamicData *>& sample_seq,
+            std::vector<dds::sub::SampleInfo *>& info_seq,
+            const rti::recording::storage::SelectorState& selector);
+
+    /*
+     * The return loan operation should free any resources allocated by the
+     * read() operation.
+     */
+    virtual void return_loan(
+            std::vector<dds::core::xtypes::DynamicData *>& sample_seq,
+            std::vector<dds::sub::SampleInfo *>& info_seq);
+
+    /*
+     * This method should flag Replay/Converter that all data related to the
+     * data stream has been read and that we're ready for termination.
+     */
+    virtual bool finished();
+
+    virtual void reset();
+
+private:
+
+    std::ifstream *data_file_;
+
```

Comment:
we should remove the blank liknes between attributes

Reply:
I will. On it.